code/modules/persistence/noticeboards.dm

```diff
@@ -61,62 +65,60 @@
 		if(!skip_icon_update)
 			update_icon()
 
-/obj/structure/noticeboard/dismantle()
+/obj/structure/noticeboard/dump_contents()
 	for(var/thing in notices)
 		remove_paper(thing, skip_icon_update = TRUE)
-	. = ..()
-
-/obj/structure/noticeboard/Destroy()
-	QDEL_NULL_LIST(notices)
-	. = ..()
+	return ..()
 
-/obj/structure/noticeboard/explosion_act(var/severity)
+/obj/structure/noticeboard/on_update_icon()
 	. = ..()
-	if(.)
-		physically_destroyed()
+	var/datum/extension/base_icon_state/bis = get_extension(src, /datum/extension/base_icon_state)
+	var/nb_notices = LAZYLEN(notices)
+	icon_state = "[bis.base_icon_state][nb_notices > 0? "0[nb_notices]" : ""]"
 
-/obj/structure/noticeboard/on_update_icon()
-	..()
-	icon_state = "[base_icon_state][LAZYLEN(notices)]"
+/obj/structure/noticeboard/handle_default_screwdriver_attackby(mob/user, obj/item/screwdriver/S)
+	var/choice = input("Which direction do you wish to place the noticeboard?", "Noticeboard Offset") as null|anything in list("North", "South", "East", "West")
+	if(choice && S.do_tool_interaction(TOOL_SCREWDRIVER, user, src, 1 SECOND))
+		default_pixel_x = 0
+		default_pixel_y = 0
+		switch(choice)
```

Comment:
Please preserve the directional setting behavior, if only for consistency with things like persistence noteboards.

Reply:
Well, the issue is that when you build one, it just floats middair. A bunch of wall mounted things are like that too. Machinery has its frame installation code taking care of it, but structures don't.

Reply:
Noticeboards should be offsetting in set_dir, it was part of a persistence fix PR I did months ago.

Reply:
Alright, I guess I can do it in set_dir.

Reply:
What do you mean actioned?
Also, I might need to get rid of that code, since wall offsets are handled differently now.